File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    if len(s) % 2 !=0:
        return 0
    stack = []
    
    for i in range(len(s)):
        if len(stack) == 0 or stack[-1] != s[i]:
            stack.append(s[i])
        elif s[i] == stack[-1]:
            stack.pop()
            

    
    
    return 1 if len(stack) == 0 else 0

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s)
{
    // stack 활용
    let stack = [];
    for(let char of s){
        if(stack.length > 0 && stack[stack.length-1] === char){
            stack.pop();
        } else{
            stack.push(char);
        }
    }
    

    return stack.length === 0 ? 1 : 0;
}